scipy/linalg/_decomp_cholesky_update.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import division, print_function, absolute_import
+import numpy
+from numpy import empty_like, sqrt, spacing, tril, triu
+from scipy.linalg import LinAlgError
+
+__all__ = ['cholesky_update']
+
+
+def cholesky_update(R, z, mod_type="+", lower=False, overwrite_R=False,
+               overwrite_z=False, check_finite=True):
+    """
+    Performs a rank-1 update or downdate of Cholesky factors.
+
+    Cholesky rank-1 update and downdate algorithm taken from [1]_ such that
+    an update results in ``D'D = R'R + zz'`` and a downdate results in ``D'D =
+    R'R - zz'``
+
+    Parameters
+    ----------
+    R : (N, N) ndarray
+        The 2D square array from which the triangular part will be used to
+        read the Cholesky factor.The remaining parts are ignored.
+    z : (N,) ndarray
+        A 1D modifier (update or downdate) array
+    mod_type: string, optional
```

Comment:
I would make this mod_type a boolean. Otherwise it's annoying to type `'-'` everytime. And I would keep a distinct name such as `downdate=False` which is pretty clear what it does.

Reply:
That might have been inspired by Matlab/Octave but I agree, it's not the most convenient choice :)